version: '3.2'
services:
  jupyterhub:
    image: jupyterhub/jupyterhub:2.3.0
    container_name: jupyterhub
    restart: always
    ports:
      - 8000:8000
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - jupyterhub_data:/srv/jupyterhub
  jupyterlab:
    image: jupyter/datascience-notebook:latest
    command: echo
  neo4j:
    image: neo4j:latest
    container_name: neo4j
    restart: always
    ports:
      - 7474:7474
      - 7687:7687
    environment:
      - NEO4J_AUTH=neo4j/SuansToy
    volumes:
      - neo4j_data:/data
      - neo4j_logs:/logs
      - neo4j_import:/var/lib/neo4j/import
      - neo4j_plugins:/plugins
  django_api:
    image: ubuntu:latest
    container_name: django
    restart: on-failure
    ports:
      - 8080:8000
    volumes:
      - django_main:/opt/services/djangoapp/src
      - django_nginx:/opt/services/djangoapp/src/config/nginx/conf.d
      - static_volume:/opt/services/djangoapp/src/static 
      - media_volume:/opt/services/djangoapp/src/media 
    depends_on:
      - postgis  
    command: ["sleep","infinity"]

volumes:
  jupyterhub_data:
  neo4j_data:
  neo4j_logs:
  neo4j_import:
  neo4j_plugins:
  django_main:
  django_nginx:
  static_volume:
  media_volume:  